### PR TITLE
Release v2.8.0

### PR DESCRIPTION
**Pull request by AurevoirXavier**

### CODE REVIEW

**Comment by AsceticBear** [comment]
Ready for review?

**Review by AsceticBear — APPROVED** [review]
The migration LGTM, some commit is a mess, I am not sure whether we should merge master again. Some already merged files are shown in the diff list. Waiting for the ci result before merging.

File: node/runtime/pangolin/src/lib.rs
Comment (HackFisher) [inline node/runtime/pangolin/src/lib.rs:943]: Would you do some recap to help me understand the upstream migration issues?, it takes me some time to catch up.

Comment (HackFisher) [reply node/runtime/pangolin/src/lib.rs:943]: Find the reason why the key does not equal.
```
[135, 141, 67, 77, 97, 37, 180, 4, 67, 254, 17, 253, 41, 45, 19, 164]    
2022-01-13 00:55:50 pre-migration prefix of storage 'Tips': 'Treasury' ==> 'Tips'    
2022-01-13 00:55:50 pre-migration prefix of storage 'Reasons': 'Treasury' ==> 'Tips'    
2022-01-13 00:55:50 panicked at 'assertion failed: new_pallet_prefix_iter.all(|key| key == storage_version_key)', /Users/denny/.cargo/git/checkouts/substrate-a7fa553ead635512/3d41798/frame/tips/src/migrations/v4.rs:125:5    
Error: Input("failed to execute 'TryRuntime_on_runtime_upgrade': Other(\"Wasm execution trapped: wasm trap: unreachable\\nwasm backtrace:\\n    0: 0xb0ee - <unknown>!rust_begin_unwind\\n\")")
```

File: node/runtime/pangolin/src/lib.rs
Comment (AsceticBear) [inline node/runtime/pangolin/src/lib.rs:941]: Can you give more explanation about why put this to the last one besides passing the check? In my mind, we should migration the storage version firstly, then do other pallets migration.

Comment (AurevoirXavier) [reply node/runtime/pangolin/src/lib.rs:941]:
```
if on_chain_storage_version < 4 { migrate } else { skip }
```

**Comment by AurevoirXavier** [comment]
```
2022-01-13 13:06:55 `pallet_timestamp::UnixTime::now` is called at genesis, invalid value returned: 0    
2022-01-13 13:06:55 [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
2022-01-13 13:06:55 [0] 💸 generated 1 npos voters, 1 from validators and 0 nominators    
2022-01-13 13:06:55 [0] 💸 new validator set of size 1 has been processed for era 1    
2022-01-13 13:06:55 [darwinia-claims] Genesis Claims List is Set to EMPTY    
2022-01-13 13:06:56 scraping key-pairs from remote @ 0x00ecbfc352f996201d9951fc64b6434ff4ae02705b540e4e5a2f42c1c81e8705    
2022-01-13 13:06:56 downloading data for all pallets.    
2022-01-13 13:09:46 injecting a total of 17619 keys    
2022-01-13 13:09:47 found matching spec name: "pangolin"    
2022-01-13 13:09:47 spec version mismatch (local 28000 != remote 27020). This could cause some issues.    
2022-01-13 13:09:47 pre-migration, prefix: 'Instance1Membership' ==> 'TechnicalMembership'    
2022-01-13 13:09:47 pre-migration prefix of storage 'Tips': 'Treasury' ==> 'Tips'    
2022-01-13 13:09:47 pre-migration prefix of storage 'Reasons': 'Treasury' ==> 'Tips'    
2022-01-13 13:09:47 pre-migration, prefix: 'Instance1Collective' ==> 'Council'    
2022-01-13 13:09:47 pre-migration, prefix: 'Instance2Collective' ==> 'TechnicalCommittee'    
2022-01-13 13:09:47 Running migration to v4 for membership with storage version <wasm:stripped>    
2022-01-13 13:09:47 migration, prefix: 'Instance1Membership' ==> 'TechnicalMembership'    
2022-01-13 13:09:47 Running migration to v4 for tips with storage version <wasm:stripped>    
2022-01-13 13:09:47 migration prefix of storage 'Tips': 'Treasury' ==> 'Tips'    
2022-01-13 13:09:47 migration prefix of storage 'Reasons': 'Treasury' ==> 'Tips'    
2022-01-13 13:09:47 Running migration to v4 for collective with storage version <wasm:stripped>    
2022-01-13 13:09:47 migration, prefix: 'Instance1Collective' ==> 'Council'    
2022-01-13 13:09:47 Running migration to v4 for collective with storage version <wasm:stripped>    
2022-01-13 13:09:47 migration, prefix: 'Instance2Collective' ==> 'TechnicalCommittee'    
2022-01-13 13:09:47 ⚠️ System declares internal migrations (which *might* execute). On-chain `<wasm:stripped>` vs current storage version `<wasm:stripped>`    
2022-01-13 13:09:47 ✅ no migration for BSC    
2022-01-13 13:09:47 ✅ no migration for Substrate2SubstrateIssuing    
2022-01-13 13:09:47 ✅ no migration for TransactionPause    
2022-01-13 13:09:47 ✅ no migration for FeeMarket    
2022-01-13 13:09:47 ✅ no migration for BridgePangoroMessages    
2022-01-13 13:09:47 ✅ no migration for BridgePangoroGrandpa    
2022-01-13 13:09:47 ✅ no migration for BridgePangoroDispatch    
2022-01-13 13:09:47 ✅ no migration for Ethereum    
2022-01-13 13:09:47 ✅ no migration for EVM    
2022-01-13 13:09:47 ✅ no migration for TronBacking    
2022-01-13 13:09:47 ✅ no migration for EthereumRelayAuthorities    
2022-01-13 13:09:47 ✅ no migration for EthereumRelayerGame    
2022-01-13 13:09:47 ✅ no migration for EthereumIssuing    
2022-01-13 13:09:47 ✅ no migration for EthereumBacking    
2022-01-13 13:09:47 ✅ no migration for EthereumRelay    
2022-01-13 13:09:47 ✅ no migration for Multisig    
2022-01-13 13:09:47 ✅ no migration for Proxy    
2022-01-13 13:09:47 ✅ no migration for Scheduler    
2022-01-13 13:09:47 ✅ no migration for Recovery    
2022-01-13 13:09:47 ✅ no migration for Society    
2022-01-13 13:09:47 ✅ no migration for Identity    
2022-01-13 13:09:47 ✅ no migration for Utility    
2022-01-13 13:09:47 ⚠️ Vesting declares internal migrations (which *might* execute). On-chain `<wasm:stripped>` vs current storage version `<wasm:stripped>`    
2022-01-13 13:09:47 ✅ no migration for Claims    
2022-01-13 13:09:47 ✅ no migration for Sudo    
2022-01-13 13:09:47 ✅ no migration for Bounties    
2022-01-13 13:09:47 ✅ no migration for Tips    
2022-01-13 13:09:47 ✅ no migration for KtonTreasury    
2022-01-13 13:09:47 ✅ no migration for Treasury    
2022-01-13 13:09:47 ✅ no migration for TechnicalMembership    
2022-01-13 13:09:47 ✅ no migration for PhragmenElection    
2022-01-13 13:09:47 ✅ no migration for TechnicalCommittee    
2022-01-13 13:09:47 ✅ no migration for Council    
2022-01-13 13:09:47 ✅ no migration for Democracy    
2022-01-13 13:09:47 ✅ no migration for HeaderMMR    
2022-01-13 13:09:47 ✅ no migration for AuthorityDiscovery    
2022-01-13 13:09:47 ✅ no migration for ImOnline    
2022-01-13 13:09:47 ✅ no migration for Grandpa    
2022-01-13 13:09:47 ✅ no migration for Session    
2022-01-13 13:09:47 ✅ no migration for Historical    
2022-01-13 13:09:47 ⚠️ Offences declares internal migrations (which *might* execute). On-chain `<wasm:stripped>` vs current storage version `<wasm:stripped>`    
2022-01-13 13:09:47 have 0 deferred offences, applying.    
2022-01-13 13:09:47 ⚠️ Staking declares internal migrations (which *might* execute). On-chain `<wasm:stripped>` vs current storage version `<wasm:stripped>`    
2022-01-13 13:09:47 ✅ no migration for ElectionProviderMultiPhase    
2022-01-13 13:09:47 ✅ no migration for Authorship    
2022-01-13 13:09:47 ✅ no migration for TransactionPayment    
2022-01-13 13:09:47 ✅ no migration for Kton    
2022-01-13 13:09:47 ✅ no migration for Balances    
2022-01-13 13:09:47 ✅ no migration for Timestamp    
2022-01-13 13:09:47 ✅ no migration for Babe    
2022-01-13 13:09:47 post-migration, prefix: 'Instance1Membership' ==> 'TechnicalMembership'    
2022-01-13 13:09:47 post-migration prefix of storage 'Tips': 'Treasury' ==> 'Tips'    
2022-01-13 13:09:47 post-migration prefix of storage 'Reasons': 'Treasury' ==> 'Tips'    
2022-01-13 13:09:47 post-migration, prefix: 'Instance1Collective' ==> 'Council'    
2022-01-13 13:09:47 post-migration, prefix: 'Instance2Collective' ==> 'TechnicalCommittee'    
2022-01-13 13:09:47 TryRuntime_on_runtime_upgrade executed without errors. Consumed weight = 2000275000000, total weight = 2000000000000 (1.0001375)
```

File: node/runtime/pangolin/src/lib.rs
Comment (AsceticBear) [reply node/runtime/pangolin/src/lib.rs:941]: Okay.  Another question: Do we have to do something in the Crab or Darwinia about storage_version migration? If so, don't forget to create an issue.

Comment (AurevoirXavier) [reply node/runtime/pangolin/src/lib.rs:941]: > Okay. Another question: Do we have to do something in the Crab or Darwinia about storage_version migration? If so, don't forget to create an issue.

https://github.com/darwinia-network/darwinia/issues/827 This one

**Review by AsceticBear — APPROVED** [review]

**Comment by AsceticBear** [comment]
RED CI

**Comment by AurevoirXavier** [comment]
> RED CI

Not sure what happen. I've re-run this 3 times. No output for build part.

**Pull request merged**
